kafka_consumer/datadog_checks/kafka_consumer/kafka_consumer.py

```diff
@@ -94,14 +94,17 @@ def check(self, instance):
 
         # Fetch the broker highwater offsets
         try:
-            self._get_highwater_offsets()
+            if len(self._consumer_offsets) < self._context_limit:
+                self._get_highwater_offsets()
+            else:
+                self.log.warning("Contex limit reached. Skipping highwater collection")
```

Comment:
```suggestion
                self.log.debug("Context limit reached. Skipping highwater offset collection.")
```

warning might be too noisy